qt/FileTreeItem.cc

```diff
@@ -188,34 +188,26 @@ QVariant FileTreeItem::data(int column, int role) const
     return value;
 }
 
-void FileTreeItem::getSubtreeWantedSize(uint64_t& have, uint64_t& total) const
+std::pair<uint64_t, uint64_t> FileTreeItem::get_subtree_wanted_size() const
 {
-    if (is_wanted_)
-    {
-        have += have_size_;
-        total += total_size_;
-    }
+    auto have = uint64_t{ is_wanted_ ? have_size_ : 0U };
+    auto total = uint64_t{ is_wanted_ ? total_size_ : 0U };
 
-    for (FileTreeItem const* const i : children_)
+    for (auto const* const child : children_)
     {
-        i->getSubtreeWantedSize(have, total);
+        auto const [child_have, child_total] = child->get_subtree_wanted_size();
+        have += child_have;
+        total += child_total;
     }
+
+    return { have, total };
 }
 
 double FileTreeItem::progress() const
 {
-    double d(0);
-    uint64_t have(0);
-    uint64_t total(0);
+    auto const [have, total] = get_subtree_wanted_size();
 
-    getSubtreeWantedSize(have, total);
-
-    if (total != 0)
-    {
-        d = static_cast<double>(have) / static_cast<double>(total);
-    }
-
-    return d;
+    return have >= total ? 1.0 : static_cast<double>(have) / static_cast<double>(total);
```

Comment:
It wouldn't hurt to avoid division by zero.
```suggestion
return have >= total ? 1.0 : (total == 0U ? 0.0 : static_cast<double>(have) / static_cast<double>(total));
```